spec/factories/registration.rb

```diff
@@ -2,6 +2,12 @@
 
 FactoryBot.define do
   factory :registration, class: WasteExemptionsEngine::Registration do
+    trait :submitted do
```

Comment:
As far as I can tell we don't actually use the sequenced reference in the new test added, and instead define a specific one when we call `create()`.

So unless I have missed something I would drop this until it's actually being used.

Reply:
My mistake! I should kill the `reference` from this trait. I will keep the `submitted_at` thought, we do use it. Without the PDF generation fails as the view expects the value to be present.